gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/TagServiceImpl.java

```diff
@@ -101,6 +101,19 @@ public TagEntity findByIdAndReference(String tagId, String referenceId, TagRefer
         }
     }
 
+    @Override
+    public void checkTagsExist(Set<String> tagIds, String referenceId, TagReferenceType referenceType) throws TechnicalException {
+        Set<Tag> foundTags = tagRepository.findByIdsAndReference(tagIds, referenceId, repoTagReferenceType(referenceType));
+        if (foundTags.size() == tagIds.size()) {
+            return;
+        }
+        foundTags.forEach(tag -> {
+            tagIds.remove(tag.getId());
+        });
```

Comment:
Maybe I miss something, by it seems that this code is useless 🤔

Reply:
We remove all the found tagIds from the original tagId list. Apparently doing a forEach of `foundTags` has a better performance than `tagIds.removeAll(foundTagIds)`.

Reply:
This is so that when we throw the TagNotFound exception, we have the list of tags not found in the message.

Reply:
oh. I didn't think about that 🙈